iio.h

```diff
@@ -928,19 +842,13 @@ __api __check_ret int iio_device_buffer_attr_write_bool(const struct iio_device
  * @param val A long long value to set the attribute to
  * @return On success, 0 is returned
  * @return On error, a negative errno code is returned */
-__api __check_ret int iio_device_buffer_attr_write_longlong(const struct iio_device *dev,
-		const char *attr, long long val);
-
-
-/** @brief Set the value of the given buffer-specific attribute
- * @param dev A pointer to an iio_device structure
- * @param attr A NULL-terminated string corresponding to the name of the
- * attribute
- * @param val A double value to set the attribute to
- * @return On success, 0 is returned
- * @return On error, a negative errno code is returned */
-__api __check_ret int iio_device_buffer_attr_write_double(const struct iio_device *dev,
-		const char *attr, double val);
+#define iio_device_buffer_attr_write(buf, attr, val)			\
+	_Generic((val),							\
+		 const char *: iio_device_buffer_attr_write_string,	\
+		 char *: iio_device_buffer_attr_write_string,		\
+		 bool: iio_device_buffer_attr_write_bool,		\
+		 long long: iio_device_buffer_attr_write_longlong,	\
+		 double: iio_device_buffer_attr_write_double)(buf, attr, val)
```

Comment:
Hmm not sure if we are being too restrictive here... While allowing for different pointer sign is arguable (personally, I would probably allow it; unless we compile with -Wpointer-sign), I would allow `unsigned long long` to be used. Not sure if it applies in some other macro

Reply:
The pointer types are different enough that it shouldn't cause a problem - I mean I wouldn't use a `void *` pointer in there.
About `long long`, here it's just a value, so the unsigned version would be implicitely cast. In the read function however it's a pointer, so the unsigned version needs to be explicitely cast.

The way I see it, is that before this PR you'd have specific API functions to read attributes, with the same types that are available here, so this PR doesn't change it. If we wanted to add support for unsigned integers here, I'd have to add an extra function (which I can still do, in a follow-up).

Reply:
> About long long, here it's just a value, so the unsigned version would be implicitely cast. In the read function however it's a pointer, so the unsigned version needs to be explicitely cast.

Did you try to run the macro with an `unsigned long long` argument? AFAIK, `_Generic` does not see signed vs unsigned as type compatible so it should fail to compile as there's mo selectable type. That is why I brought this up...

Reply:
For `unsigned long long` I guess it's possible to add an entry with the same `iio_device_buffer_attr_write_longlong`, and the argument will be cast implicitely.
(Some compilers like MSVC might warn about that though...)

Reply:
> For unsigned long long I guess it's possible to add an entry with the same iio_device_buffer_attr_write_longlong, and the argument will be cast implicitely.

Yes, that is what I had in mind... No idea about the MSVC compilers though